pkg/kubectl/rolling_updater.go

```diff
@@ -763,6 +759,33 @@ func updateWithRetries(rcClient client.ReplicationControllerInterface, rc *api.R
 	return rc, err
 }
 
+type updateFuncForPod func(pod *api.Pod)
+
+// updateWithRetriesForPod updates applies the given pod as an update.
+func updateWithRetriesForPod(podClient client.PodInterface, pod *api.Pod, applyUpdate updateFuncForPod) (*api.Pod, error) {
```

Comment:
Since you are updating meta in both cases, you could have a generic `updateWithRetries` by using a meta accessor, no?